Documentation/proposals/202405-scrapeconfig-graduation.md

```diff
@@ -0,0 +1,213 @@
+
+## How
+
+In order to provide a more comprehensive and versatile monitoring solution, enhancing the
+Service Discovery support in the Prometheus Operator is crucial. This will address current
+limitations and better serve diverse user needs. The following steps outline our approach to achieving this goal:
+
+#### Additional Service Discovery Configurations
+
+By adding more Service Discovery configurations, we increase the flexibility and utility of the CRD for various user scenarios.
+Kubernetes being the sole target for Prometheus-Operator, we think it is fitting to provide complete support for Kubernetes service discovery, this includes
+supporting all fields present in the Prometheus configuration, example use cases and keeping the Service Discovery robustly maintained in general. We believe that the Kubernetes service discovery
+along with the other existing service discoveries that we offer should be sufficient for most user's needs and so implementing all the Service discoveries
+is not a priority. However, this does not imply service discovery support is limited to what exists today.
+It would be a better use of time to add the service discoveries that the users need as we get feature requests for them.
+
+The following is the list of Service Discoveries which we want to support before graduation:
+
+- *`azure_sd_config`*
+- *`consul_sd_config`*
+- *`digitalocean_sd_config`*
+- *`docker_sd_config`*
+- *`dockerswarm_sd_config`*
+- *`dns_sd_config`*
+- *`ec2_sd_config`*
+- *`openstack_sd_config`*
+- *`puppetdb_sd_config`*
+- *`file_sd_config`*
+- *`gce_sd_config`*
+- *`hetzner_sd_config`*
+- *`http_sd_config`*
+- *`kubernetes_sd_config`*
+- *`kuma_sd_config`*
+- *`lightsail_sd_config`*
+- *`linode_sd_config`*
+- *`nomad_sd_config`*
+- *`eureka_sd_config`*
+- *`ovhcloud_sd_config`*
+- *`scaleway_sd_config`*
+- *`ionos_sd_config`*
+
+If we were to categorize the Service Discoveries based on the amount of effort we are willing to put in maintaining them:
+
+**Tier-1:**
+Deploying the Operator to Kubernetes is our sole use case. So it is logical that we want to provide maximum support to any service discovery that fundamentally fits into the a Kubernetes cluster.
+- Kubernetes Service Discovery
+- File Service Discovery
+- Static config
+- DNS Service Discovery
+- HTTP Service Discovery
+
+**Tier-2:**
+Deploying the Operator to Kubernetes is our sole use case, so it is logical to extend support for cloud providers who provide a managed Kubernetes service.
+- Azure Service Discovery
+- EC2 Service Discovery
+- Lightsail Service Discovery
+- DigitalOcean Service Discovery
+- Kuma Service Discovery
+- OVHCloud Service Discovery
+- Scaleway Service Discovery
+- Ionos Service Discovery
+- OpenStack Service Discovery
+Though it doesn't fit with the description given above, Consul is a widely used solution. For this reason, it makes sense to include Consul Service Discovery in this list.
+- Consul Service Discovery
+
```

Comment:
Maybe we should say here that the project maintainers don't commit on anything for the other service discoveries not listed above?

Reply:
Agreed.